rust/protocol/src/protocol.rs

```diff
@@ -625,7 +632,7 @@ mod tests {
     use rand::rngs::OsRng;
     use rand::{CryptoRng, Rng};
 
-    fn create_signal_message<T>(csprng: &mut T) -> SignalMessage
+    fn create_signal_message<T>(csprng: &mut T) -> Result<SignalMessage>
```

Comment:
This one doesn't really make sense to me. It may be a helper function for a test, but it actually cannot fail, and immediately panicking is valid. Can you use `expect` instead?

Reply:
Sure I guess. Reasoning was to avoid as many `expects` in `src` as possible since those need to be audited for panics, and having tests using unwrap/expect makes it harder to find ones that are in the protocol src proper.

Reply:
Eh, my opinion is that auditing is all well and good, but not at the cost of actual type signatures. I can see how being a test helper puts it on the edge, though.

Another option would be to go *back* to use `unwrap` in tests, since the Clippy warning keeps it out of the src files. Any of these three options are all right with me though.